pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -127,18 +134,77 @@ func (s *requested) Execute(msg dispatcher.DIDCommMsg) (state, error) {
 			return nil, fmt.Errorf("outbound invitations are not allowed for state %s", s.Name())
 		}
 		// send did-exchange Request
-		return &noOp{}, nil
+		return sendExchangeRequest(msg)
 	case connectionRequest:
 		if msg.Outbound {
 			// send outbound Request
-			return &noOp{}, nil
+			return sendOutboundrequest(msg)
 		}
 		return &responded{}, nil
 	default:
 		return nil, fmt.Errorf("illegal msg type %s for state %s", msg.Type, s.Name())
 	}
 }
 
+func sendExchangeRequest(msg dispatcher.DIDCommMsg) (state, error) {
+	id := uuid.New()
+	//prepare the request
+	request := &Request{
```

Comment:
this request is missing a `Connection` object

Reply:
added